noether/src/segmentation_server.cpp

```diff
@@ -36,17 +36,39 @@ class SegmentationAction
   void executeCB(const noether_msgs::SegmentGoalConstPtr& goal)
   {
     // Step 1: Load parameters
-    // TODO: Pass these in along with action
-    std::string file;
-    double curvature_threshold;
-    int min_cluster_size, max_cluster_size;
-    bool show_individually, save_outputs;
-    pnh_.param<std::string>("filename", file, "");
-    pnh_.param<int>("min_cluster_size", min_cluster_size, 500);
-    pnh_.param<int>("max_cluster_size", max_cluster_size, 1000000);
-    pnh_.param<double>("curvature_threshold", curvature_threshold, 0.3);
-    pnh_.param<bool>("show_individually", show_individually, false);
-    pnh_.param<bool>("save_outputs", save_outputs, false);
+    double curvature_threshold = goal->segmentation_config.curvature_threshold;
+    int min_cluster_size = goal->segmentation_config.min_cluster_size;
+    int max_cluster_size = goal->segmentation_config.max_cluster_size;
+
+    int windowed_sinc_iterations = goal->filtering_config.windowed_sinc_iterations;
+    double windowed_sinc_pass_band = goal->filtering_config.windowed_sinc_pass_band;
+    double windowed_sinc_feature_angle = goal->filtering_config.windowed_sinc_edge_angle;
+    double windowed_sinc_edge_angle = goal->filtering_config.windowed_sinc_edge_angle;
+    bool windowed_sinc_edge_smoothing = goal->filtering_config.windowed_sinc_edge_smoothing;
+    bool windowed_sinc_boundary_smoothing = goal->filtering_config.windowed_sinc_boundary_smoothing;
+    bool windowed_sinc_nonmanifold_smoothing = goal->filtering_config.windowed_sinc_nonmanifold_smoothing;
+    bool windowed_sinc_normalize_coordinates = goal->filtering_config.windowed_sinc_normalize_coordinates;
+
+    int laplacian_iterations = goal->filtering_config.laplacian_iterations;
+    double laplacian_relaxation_factor = goal->filtering_config.laplacian_relaxation_factor;
+    double laplacian_edge_angle = goal->filtering_config.laplacian_edge_angle;
+    bool laplacian_edge_smoothing = goal->filtering_config.laplacian_edge_smoothing;
+    bool laplacian_boundary_smoothing = goal->filtering_config.laplacian_boundary_smoothing;
```

Comment:
It seems like you could have use a reference variable for goal->filtering_config with a rather short name and use that reference to access each individual member without having to type the full name.  In short, I would've done as follows
```
auto& cfg = goal->filtering_config;
// and then 
double curvature_threshold = (cfg.curvature_threshold < 0.0001) ? 0.05 : cfg.curvature_threshold;
int min_cluster_size = (cfg.min_cluster_size ==0) ? 500 : cfg.min_cluster_size;
.
.
.
```